### PR TITLE
[ASDataController] Prevent calling endUpdatesAnimated:completion: in an unbalanced way

### PR DESCRIPTION
**Pull request by maicki**

### CODE REVIEW

**Comment by Adlai-Holler** [comment]
Hmm this change seems good, and it exposed some other (apparently serious) internal logic issue. Even better! I've looked at the code for those failing tests and they use the API correctly, so the failure may lie in ASTableView -> ASChangeSetDataController. I will look at this today and see if that fix is small enough to address in this PR. If not, we will create a separate issue and temporarily disable these tests.

File: AsyncDisplayKit/Details/ASChangeSetDataController.m [hunk 1]
```diff
@@ -43,6 +40,9 @@ - (void)endUpdatesAnimated:(BOOL)animated completion:(void (^)(BOOL))completion
   ASDisplayNodeAssertMainThread();
   _changeSetBatchUpdateCounter--;
   
+  // Prevent calling endUpdatesAnimated:completion: in an unbalanced way
+  NSAssert(_changeSetBatchUpdateCounter < 0, @"endUpdatesAnimated:completion: called without having a balanced beginUpdates call");
```
Comment (Adlai-Holler) [inline AsyncDisplayKit/Details/ASChangeSetDataController.m]: Let's do `_changeSetBatchUpdateCount >= 0` instead

**Comment by Adlai-Holler** [comment]
@maicki Once the assertion is flipped, this is ready to 🚢

**Comment by maicki** [comment]
@Adlai-Holler Ready to roll 🚗

**Pull request merged**